### DISCUSSION TITLE
Where device.js is?

### DISCUSSION THREAD

**Discussion by hoattt1** [Koenkk/zigbee2mqtt#5309] [General]
Hi everyone
I am a newbie and watn to be add some Tuya switch 4gang - zigbee3.0. But when i connect it to hassOS using Zigbee2mqtt, it shows unsupported device. I read the instruction in "https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html" but i can't find where device.js is.
My system:
 _ raspberry pi 3b+
 _ HassOS

Thanks!

**Comment by sjorge** [comment]
It's located in  this repo https://github.com/Koenkk/zigbee-herdsman-converters

Locally you can find it in node_modules/zigbee-herdsman-converters, do note updating z2m will replace the file again.

**Reply by GD-Dal** [reply]
Hi.
I'm wondering this myself. I use Home Assistant OS 5.9 as Operating system, and can't seem to find those files anywhere.
